INT: don't generate useless match arm for irrefutable patterns in IfLetToMatchIntention

Kobzol opened the pull request:
This PR modifies `IfLetToMatchIntention` to avoid generating useless `_ => { ... }` arms if the match arm is already irrefutable. Before the generated code would right away contain a warning that the arms after the irrefutable arm can be removed.

It also removes the code from a potential else branch if the pattern is irrefutable, since the else was basically dead code.

mchernyavsky commented on src/test/kotlin/org/rust/ide/intentions/IfLetToMatchIntentionTest.kt:
```diff
@@ -636,4 +636,47 @@ class IfLetToMatchIntentionTest : RsIntentionTestBase(IfLetToMatchIntention()) {
+        }
+    """)
+
+    fun `test irrefutable pattern with else`() = doAvailableTest("""
```
Hmmm, I think we should keep else-branch here. @Undin what do you think?

Undin replied:
I agree that we should keep else branch if a user already has it

Undin replied:
BTW, it would be great to have an inspection to annotate such redundant `else`

Undin commented on src/test/kotlin/org/rust/ide/intentions/IfLetToMatchIntentionTest.kt:
```diff
@@ -636,4 +636,47 @@ class IfLetToMatchIntentionTest : RsIntentionTestBase(IfLetToMatchIntention()) {
+        struct S { a: Id, b: u32 }
+        fn foo(s: S) {
+            if let S { a: Id(ref name), .. } = s/*caret*/ {
+                let _x = a;
```
```suggestion
                let _x = name;
```

Otherwise, this code cannot be compiled because `a` here is not binding
Fix it please in other tests as well

Kobzol commented:
Ok, I changed the `else` branch behaviour and rebased over master. Should I create (in a separate PR) a new `RsLocalInspectionTool` similar to `RsMissingElseInspection` (possibly with a quick fix to remove the `else` branch) or put the inspection elsewhere?

Undin commented:
> Should I create (in a separate PR) a new RsLocalInspectionTool similar to RsMissingElseInspection (possibly with a quick fix to remove the else branch) or put the inspection elsewhere?

@Kobzol I'd prefer to make it a separate inspection

Undin commented:
bors retry
